contracts/strategies/StrategyBase.sol

```diff
@@ -0,0 +1,298 @@
+  ) VaultAdapter() {
+    vault = _vault;
+    optionType = _optionType;
+    gwavOracle = _gwavOracle;
+  }
+
+  function initAdapter(
+    address _curveSwap,
+    address _optionToken,
+    address _optionMarket,
+    address _liquidityPool,
+    address _shortCollateral,
+    address _synthetixAdapter,
+    address _optionPricer,
+    address _greekCache,
+    address _quoteAsset,
+    address _baseAsset,
+    address _feeCounter
+  ) external onlyOwner {
+    // set addressese for LyraVaultAdapter
+    setLyraAddresses(
+      _curveSwap,
+      _optionToken,
+      _optionMarket,
+      _liquidityPool,
+      _shortCollateral,
+      _synthetixAdapter,
+      _optionPricer,
+      _greekCache,
+      _quoteAsset,
+      _baseAsset,
+      _feeCounter
+    );
+
+    quoteAsset.approve(address(vault), type(uint).max);
+    baseAsset.approve(address(vault), type(uint).max);
+    collateralAsset = _isBaseCollat() ? baseAsset : quoteAsset;
+  }
+
+  /**
+   * @dev update the base strategy for the new round.
+   */
+  function setBaseStrategy(BaseStrategy memory _baseStrategy) external onlyOwner {
+    (, , , , , , , bool roundInProgress) = vault.vaultState();
+    require(!roundInProgress, "cannot change strategy if round is active");
+    baseStrategy = _baseStrategy;
+  }
+
+  ///////////////////
+  // VAULT ACTIONS //
+  ///////////////////
+
+  /**
+   * @dev set the board id that will be traded for the next round
+   * @param boardId lyra board Id.
+   */
+  function _setBoard(uint boardId) internal {
+    Board memory board = getBoard(boardId);
+    require(_isValidExpiry(board.expiry), "invalid board");
+    activeExpiry = board.expiry;
+  }
+
+  /**
+   * @dev exchange asset back to collateral asset and send it back to the vault
+   */
+  function _returnFundsToVaut() internal {
+    ExchangeRateParams memory exchangeParams = getExchangeParams();
+    uint quoteBal = quoteAsset.balanceOf(address(this));
+
+    if (_isBaseCollat()) {
+      // exchange quote asset to base asset, and send base asset back to vault
+      uint baseBal = baseAsset.balanceOf(address(this));
+      uint minQuoteExpected = quoteBal.divideDecimal(exchangeParams.spotPrice).multiplyDecimal(
+        DecimalMath.UNIT - exchangeParams.baseQuoteFeeRate
+      );
+      uint baseReceived = exchangeFromExactQuote(quoteBal, minQuoteExpected);
+      require(baseAsset.transfer(address(vault), baseBal + baseReceived), "failed to return funds from strategy");
+    } else {
+      // send quote balance directly
+      require(quoteAsset.transfer(address(vault), quoteBal), "failed to return funds from strategy");
+    }
+  }
+
+  ////////////////
+  // Validation //
+  ////////////////
+
+  /**
+   * @dev verify if the strike is valid for the strategy
+   * @return isValid true if vol is withint [minVol, maxVol] and delta is within targetDelta +- maxDeltaGap
+   */
+  function isValidStrike(Strike memory strike) public view returns (bool isValid) {
```

Comment:
same as L37 comment, maybe just make this a virtual function?

Reply:
removed from base strategy.